plaso/engine/artifacts_filter_file.py

```diff
@@ -0,0 +1,272 @@
+    artifact_registry = artifacts_registry.ArtifactDefinitionsRegistry()
+    artifact_reader = artifacts_reader.YamlArtifactsReader()
+
+    try:
+      artifact_registry.ReadFromFile(artifact_reader, self._path)
+
+    except (KeyError, artifacts_errors.FormatError) as exception:
+      raise errors.BadConfigOption((
+          'Unable to read artifact definitions from: {0:s} with '
+          'error: ''{1!s}').format(self._path, exception))
+
+    for definition in artifact_registry.GetDefinitions():
+      for source in definition.sources:
+        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+          for path_entry in source.paths:
+            self.BuildFindSpecsFromFileArtifact(path_entry, source.separator,
+                                                path_attributes, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(key_entry,
+                                                      path_attributes,
+                                                      find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
+          logging.warning(('Unable to handle Registry Value, extracting '
+                           'key only: {0:s} ').format(source.key_value_pairs))
+
+          for key_pair in source.key_value_pairs:
+            keys = set()
+            keys.add(key_pair.get('key'))
+            for key_entry in keys:
+              if self._CheckKeyCompatibility(key_entry):
+                self.BuildFindSpecsFromRegistryArtifact(key_entry,
+                                                        path_attributes,
+                                                        find_specs)
+        else:
+          logging.warning(('Unable to handle artifact, plaso does not '
+                           'support: {0:s} ').format(source.type_indicator))
+    self._knowledge_base.SetValue(ARTIFACTS_FILTER_FILE, find_specs)
+
+  def BuildFindSpecsFromFileArtifact(self, path_entry, separator,
+                                     path_attributes, find_specs):
+    """Build find specification from a forensic artifacts file.
+
+    Args:
+      path_entry (str):  Current file system path to add.
+          environment variables.
+      separator (str): File system path separator.
+      path_attributes list(str):  Environment variable attributes used to
+          dynamically populate environment variables in key.
+      find_specs dict[artifacts.artifact_types]:  Dictionary containing
+          find_specs.
+    """
+    for path in self._ExpandGlobs(path_entry):
+      if path_attributes:
+        try:
+          if '%%environ_' in path:
```

Comment:
make this more specific only known %%...%% should be replaced, also don't we do this already elsewhere in the codebase? maybe refactor to reuse existing functionatlity

Reply:
I looked through trying to find where we might doing this already, but couldn't find it.  Can you point me towards it or should I implement?

Reply:
I think I found and modified the path_helper to assist with this.  There's some more work that could be done to parse more variable like structures such as users.localappdata.